src/NServiceBus.RabbitMQ/Configuration/ConnectionStringParser.cs

```diff
@@ -39,63 +40,51 @@ public ConnectionConfiguration Parse(string connectionString)
 
             if (ContainsKey("host"))
             {
-                ParseHosts(connectionConfiguration, this["host"] as string);
+                ParseHosts(connectionConfiguration, this["host"] as string, invalidOptionsMessage);
             }
             else
             {
-                throw new Exception("Invalid connection string. 'host' value must be supplied. e.g: \"host=myServer\"");
+                invalidOptionsMessage.AppendLine("Invalid connection string. 'host' value must be supplied. e.g: \"host=myServer\"");
             }
 
             if (ContainsKey("dequeuetimeout"))
             {
-                var message = "The 'DequeueTimeout' connection string option has been removed. Consult the documentation for further information.";
-
-                Logger.Error(message);
-
-                throw new NotSupportedException(message);
+                invalidOptionsMessage.AppendLine("The 'DequeueTimeout' connection string option has been removed. Consult the documentation for further information.");
             }
 
             if (ContainsKey("maxwaittimeforconfirms"))
             {
-                var message = "The 'MaxWaitTimeForConfirms' connection string option has been removed. Consult the documentation for further information";
-
-                Logger.Error(message);
-
-                throw new NotSupportedException(message);
+                invalidOptionsMessage.AppendLine("The 'MaxWaitTimeForConfirms' connection string option has been removed. Consult the documentation for further information");
             }
 
             if (ContainsKey("prefetchcount"))
             {
-                var message = "The 'PrefetchCount' connection string option has been removed. Use 'EndpointConfiguration.UseTransport<RabbitMQTransport>().PrefetchCount' instead.";
-
-                Logger.Error(message);
-
-                throw new NotSupportedException(message);
+                invalidOptionsMessage.AppendLine("The 'PrefetchCount' connection string option has been removed. Use 'EndpointConfiguration.UseTransport<RabbitMQTransport>().PrefetchCount' instead.");
             }
 
             if (ContainsKey("usepublisherconfirms"))
             {
-                var message = "The 'UsePublisherConfirms' connection string option has been removed. Use 'EndpointConfiguration.UseTransport<RabbitMQTransport>().UsePublisherConfirms' instead.";
+                invalidOptionsMessage.AppendLine("The 'UsePublisherConfirms' connection string option has been removed. Use 'EndpointConfiguration.UseTransport<RabbitMQTransport>().UsePublisherConfirms' instead.");
+            }
 
-                Logger.Error(message);
+            if (invalidOptionsMessage.Length > 0)
+            {
+                Logger.Error(invalidOptionsMessage.ToString());
```

Comment:
Not on the hot path, so it doesn't matter a great deal, but would it better to store the `ToString` result and reuse it instead of allocating two strings?